### PR TITLE
add ui injection

### PR DESCRIPTION
**Pull request by yhattav**

### CODE REVIEW

File: packages/gallery/src/components/item/videos/videoItemWrapper.tsx [hunk 1]
```diff
@@ -20,12 +20,17 @@ class VideoItemImplementation extends React.Component<MediaImplementationProps>
     const shouldForceLoadVideo = useTransparentPlayButtonAndForceLoadVideo(
       this.props
     );
+    console.log(this.props.showPlayButton, 'showPlayButton');
```
Comment (shmuelhizmi) [inline packages/gallery/src/components/item/videos/videoItemWrapper.tsx]: hello

File: packages/gallery/src/context/GalleryContext.js [hunk 2]
```diff
@@ -1,23 +1,19 @@
 import React from 'react';
 
-export let GalleryContext;
-try {
-  GalleryContext = React.createContext({});
-} catch (e) {
-  GalleryContext = null;
-}
+const GalleryContext = React.createContext({});
+console.log(
```
Comment (shmuelhizmi) [inline packages/gallery/src/context/GalleryContext.js]: :)))

File: packages/gallery/src/components/item/media/GalleryUI.tsx [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+interface GalleryUIProps {
+  size: number;
+  type: 'videoPlayButton' | 'rotateArrow';
+}
+
+export const GalleryUI = ({ type, size }: GalleryUIProps): JSX.Element => {
+  let Component;
+
+  const galleryUI = useGalleryUI();
+  if (typeof galleryUI?.[type] === 'function') {
```
Comment (liatv) [inline packages/gallery/src/components/item/media/GalleryUI.tsx:20]: why would galleryUI.type would be a function

File: packages/gallery/src/context/GalleryContext.js (hunk 2 above)
Comment (shmuelhizmi) [reply packages/gallery/src/context/GalleryContext.js]: what about something like
```tsx
import React from "react";

export function createComponentProvider<
  T extends Record<string, React.ComponentType<any>>
>(defaultComponents: T) {
  const context = React.createContext<T>(defaultComponents);
  function OverrideUI({
    components,
    children,
  }: {
    components: Partial<T>;
    children: React.ReactNode;
  }) {
    return (
      <context.Consumer>
        {(parentComponents) => (
          <context.Provider value={{ ...parentComponents, ...components }}>
            {children}
          </context.Provider>
        )}
      </context.Consumer>
    );
  }
  const components = Object.keys(defaultComponents).reduce(
    (acc, key) => ({
        ...acc,
        [key]: (props: any) => (
            <context.Consumer>
                {(components) => {
                    const Component = components[key];
                    return <Component {...props} />;
                }}
            </context.Consumer>
        ),
    }),
    {}
    ) as T;

    return { OverrideUI, components };
}


// Usage

const { OverrideUI, components } = createComponentProvider({
    Button: ({ children }) => <button>{children}</button>,
    Input: ({ children }) => <input value={children} />,
});

const { Button, Input } = components;

export {
    OverrideUI,
    Button,
    Input,
}

const App = () => (
    <OverrideUI components={{ Button: ({ children }) => <button>{children}!</button> }}>
        <Button>Click me</Button>
        <Input>Some text</Input>
    </OverrideUI>
);
```

**Review by shmuelhizmi — CHANGES_REQUESTED** [review]

File: packages/gallery/src/context/GalleryContext.js (hunk 2 above)
Comment (liatv) [reply packages/gallery/src/context/GalleryContext.js]: KING

File: packages/gallery/src/components/item/media/GalleryUI.tsx (hunk 3 above)
Comment (yhattav) [reply packages/gallery/src/components/item/media/GalleryUI.tsx:20]: galleryUI is an object of custom components

File: packages/gallery/src/context/GalleryContext.js (hunk 2 above)
Comment (yhattav) [reply packages/gallery/src/context/GalleryContext.js]: I'll push this and then we can refactor it further. I like the fact that the components can be imported by name and not a prop (type).

**Pull request merged**